src/hooks/useTimeout.js

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2019 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+import { useCallback, useEffect, useRef } from 'react'
+
+/**
+ * Set a timer, which executes a function once the timer expires. Exposes pause, resume and cancel
+ * callbacks to the consumers.
+ *
+ * @param fn Callback function to execute after the specified timeout
+ * @param ms Timeout in milliseconds after which to execute the callback
+ */
+export const useTimeout = (fn, ms) => {
+  // Indicates if timer is currently running
+  const isRunning = useRef(false)
+
+  // Number of milliseconds remaining before the timer runs out and callback is executed
+  const msRemaining = useRef(ms)
+
+  // Time when the most recent execution was requested (0 if time is not currently running)
+  const timeStarted = useRef(0)
+
+  // Timeout handle
+  const handle = useRef(0)
+
+  // Original callback function
+  const callback = useRef(fn)
```

Comment:
don't use names such as `fn`, `ms`, `handle`, `callback`, etc.

Reply:
@mavdryk since it's a hook that exports readable methods, I'm fine with keeping these consts names as is